agent_build_refactored/managed_packages/managed_packages_builders.py

```diff
@@ -0,0 +1,932 @@
+        :param token: Auth token to Packagecloud.
+        :param repo_name: Target Packagecloud repo.
+        :param user_name: Target Packagecloud user.
+        """
+
+        # Run in docker if needed.
+        if self.runs_in_docker:
+            self.run_in_docker(
+                command_args=[
+                    "publish",
+                    "--packages-dir",
+                    RunnerMappedPath(packages_dir_path),
+                    "--token",
+                    token,
+                    "--repo-name",
+                    repo_name,
+                    "--user-name",
+                    user_name
+                ]
+            )
+            return
+
+        # Set packagecloud's credentials file.
+        config = {
+            "url": "https://packagecloud.io",
+            "token": token
+        }
+
+        config_file_path = pl.Path.home() / ".packagecloud"
+        config_file_path.write_text(
+            json.dumps(config)
+        )
+
+        packages_to_publish_file = packages_dir_path / "packages_to_publish.json"
+        packages_to_publish = set(json.loads(
+            packages_to_publish_file.read_text()
+        ))
+
+        for package_path in packages_dir_path.glob(f"*.{self.PACKAGE_TYPE}"):
+            if package_path.name not in packages_to_publish:
+                logger.info(f"Package {package_path.name} has been skipped since it's already in repo.")
+                continue
+
+            repo_paths = {
+                "deb": f"{user_name}/{repo_name}/any/any",
+                "rpm": f"{user_name}/{repo_name}/rpm_any/rpm_any",
+            }
+
+            check_call_with_log(
+                [
+                    "package_cloud",
+                    "push",
+                    repo_paths[self.PACKAGE_TYPE],
+                    str(package_path)
+                ]
+            )
+
+            logging.info(f"Package {package_path.name} is published.")
+
+    def _search_for_packages_in_repo(
+            self,
+            params: Dict,
+            token: str,
+            user_name: str,
+            repo_name: str,
+    ) -> List:
+        """
+        Search packages with the given query params in the Packagecloud repository.
+        :param params: Params for the url query.
+        :param token: Packagecloud token
+        :param repo_name: Target Packagecloud repo.
+        :param user_name: Target Packagecloud user.
+        :return: List of found packages.
+        """
+        import requests
+        from requests.auth import HTTPBasicAuth
+
+        auth = HTTPBasicAuth(token, "")
+
+        param_filters = {
+            "deb": "debs",
+            "rpm": "rpms"
+        }
+
+        params["filter"] = param_filters[self.PACKAGE_TYPE]
+
+        with requests.Session() as s:
+            resp = s.get(
+                url=f"https://packagecloud.io/api/v1/repos/{user_name}/{repo_name}/search.json",
```

Comment:
EDIT: It looks like it does - https://packagecloud.io/docs/api#pagination

So we likely either need to handle this scenario (in case the filter results in more than one page of results being returned) or simply throw in case we encounter such condition (and we can tackle it later once more packages are in the repo or similar).

Initially we may also want to use larger value for ``?per_page`` query parameter.